crates/ruff_python_parser/tests/snapshots/invalid_syntax@statements__with__ambiguous_lpar_with_items.py.snap

```diff
@@ -1360,7 +1363,7 @@ Module(
 2 | # These cases should raise the correct syntax error and recover properly.
 3 | 
 4 | with (item1, item2),: ...
-  |                     ^ Syntax Error: Expected an expression
+  |                    ^ Syntax Error: Trailing comma not allowed
 5 | with (item1, item2), as f: ...
```

Comment:
I haven't included this logic (https://github.com/astral-sh/ruff/blob/9b2cf569b22855439fa916be6fc417b372074f42/crates/ruff_python_parser/src/parser/statement.rs#L1910-L1940) which took care of raising "expected an expression" error. It's an edge case where we would raise a different error while in the following we'd raise "trailing comma not allowed".

```py
with (item1, item2), item3,: ...
```

Reply:
I couldn't find a simple way to do it and I'm not sure if it's worth it. The "trailing comma" error is raised by list parsing while the "expect an expression" is raised by the with-item parsing logic. Even if I'm able to add the logic, both errors will be displayed because they're raised at separate location but they both contradict each other. So, we should only raise one of them.

Reply:
I think both errors are equally good and ask the same fo the user. Remove the comma or add an expression.